src/ZoneServer/World/Actors/Characters/Character.cs

```diff
@@ -594,6 +609,9 @@ public void Warp(int mapId, Position pos)
 			if (!ZoneServer.Instance.Data.MapDb.TryFind(mapId, out var map))
 				throw new ArgumentException("Map '" + mapId + "' not found in data.");
 
+			if (this.IsOutOfBody())			
```

Comment:
I feel like this is a weird place to check for out of body to prevent warps, because what if we do want to warp? This special behavior also isn't documented anywhere, so anyone who might try to warp someone will have to dig deep into the warp code to figure out why it's not working.

Additionally, creating explicit checks for very specific conditions within general purpose functions is not ideal, because it's not extensible. An alternative idea that I'm just going to throw out there might be to add a warp lock, which you can set for these buffs you're checking in `IsOutOfBody`, which will generalize the whole approach, make it somewhat self-documenting, save you from having to do that manual buff check, and have it be reusable.

Reply:
You marked this as resolved, but the issues are still there 🤔